libraries/chain/chain_controller.cpp

```diff
@@ -367,8 +367,12 @@ transaction_trace chain_controller::delayed_transaction_processing( const transa
 
    time_point_sec execute_after = head_block_time();
    execute_after += mtrx.delay;
-   //TODO: !!! WHO GETS BILLED TO STORE THE DELAYED TX?
-   deferred_transaction dtrx(context.get_next_sender_id(), config::system_account_name, config::system_account_name, execute_after, trx);
+
+   // TODO: update to better method post RC1?
+   FC_ASSERT(mtrx.trx().actions.size() > 0 && mtrx.trx().actions.at(0).authorization.size() > 0, "Delayed transactions must have at least one action and those must have at least one authorization");
```

Comment:
This is imposing stricter requirements than our current requirement that at least one authorization must exist for some action in the transaction.

Instead, you could search through the actions for the first action with authorizations, and then choose the actor of the first authorizer of that action.